src/components/dropdownSearch/DropdownSearch.tsx

```diff
@@ -0,0 +1,244 @@
+    if (!_.isEmpty(this.props.filterText)) {
+      const textFilterElements: (JSX.Element | string)[] = [''];
+      index = text.toLowerCase().indexOf(this.props.filterText.toLowerCase());
+      while (index !== -1) {
+        if (index > 0) {
+          textFilterElements.push(text.substring(0, index));
+        }
+        textFilterElements.push(this.getTextElement(text.substring(index, index + this.props.filterText.length), originalText, 'bold'));
+        text = text.substring(index + this.props.filterText.length);
+        index = text.toLowerCase().indexOf(this.props.filterText.toLowerCase());
+      }
+      textFilterElements.push(text);
+
+      this.textIndex = 0;
+      return textFilterElements;
+    }
+
+    return text;
+  }
+
+  private getTextElement(subText: string, text: string, className: string = ''): JSX.Element {
+    this.textIndex = this.textIndex + 1;
+    return <span key={`${text}-${this.textIndex}`} className={className}>{subText}</span>;
+  }
+
+  private getDropdownPrepend(option: IDropdownOption): JSX.Element {
+    if (option && option.prefix) {
+      return <span key={option.prefix} className='dropdown-prepend'>{option.prefix}</span>;
+    }
+    return null;
+  }
+
+  private getMainInput(): JSX.Element {
+    if (this.props.isOpened) {
+      return <FilterBox id={this.props.id}
+        onFilter={(id, filterText) => this.handleOnFilter(id, filterText)}
+        onBlur={() => this.handleOnBlur()}
+        filterPlaceholder={this.props.filterPlaceholder || this.props.selectedOption.displayValue ||
+          this.props.selectedOption.value}
+        isAutoFocus={true} />;
+    } else {
+      return <button className='btn dropdown-toggle dropdown-button-search-container mod-search' type='button' data-toggle='dropdown'
+        onClick={() => this.handleOnClick()}
+        style={{
+          maxWidth: this.props.maxWidth,
+        }}
+        disabled={!!this.props.isDisabled}>
+        {this.getDropdownPrepend(this.props.selectedOption)}
+        {this.getSvg(this.props.selectedOption)}
+        {this.getSelectedOption()}
+        <span className='dropdown-toggle-arrow'></span>
+      </button>;
+    }
+  }
+
+  handleOnFilter(id: string, filterText: string) {
+    if (this.props.onFilterClick) {
+      this.props.onFilterClick(filterText);
+    }
+  }
+
+  handleOnClick() {
+    if (this.props.onClick) {
+      this.props.onClick();
+    }
+
+    if (this.props.onToggleDropdown) {
+      this.props.onToggleDropdown();
+    }
+  }
+
+  handleOnOptionClick = (e: any) => {
+    if (e.target && this.props.onOptionClick) {
+      const option = _.findWhere(this.props.options, { value: e.currentTarget.dataset.value });
+      this.props.onOptionClick(option);
+    }
+  }
+
+  handleOnBlur() {
+    if (this.props.onBlur) {
+      this.props.onBlur();
+    }
+  }
+
+  componentWillMount() {
+    if (this.props.onRender) {
+      this.props.onRender(this.props.id);
```

Comment:
The ID is useless here.
You can define `onRender` to have type `() => void`.

In your `mapDispatchToProps` function, you can do `onRender: () => dispatch(someAction(ownProps.id))`

Same for `onMount`, but I believe you can merge both functions.